### PR TITLE
update spec of ECID

### PR DESCRIPTION
**Pull request by mraszyk**
Update effective canister ID specification.

### CODE REVIEW

File: spec/index.adoc [hunk 1]
```diff
@@ -594,26 +594,25 @@ Canister methods that do not change the canister state can be executed more effi
...
+
+NOTE: The expectation is that user-side agent code shields users and developers from the notion of an effective canister ID, in analogy to how the System API interface shields canister developers from worrying about routing.
 
 +
 [NOTE]
 ====
-The Internet Computer blockchain mainnet does not support `provisional_create_canister_with_cycles`. This means that using an effective canister id that could be an existing canister would lead to the request being routed to a node on the corresponding subnet and produce an error response there, while using an effective canister id that cannot be an existing canister id would cause an error response from the node receiving the request -- either is fine.
+The Internet Computer blockchain mainnet does not support `provisional_create_canister_with_cycles`. This means that using an effective canister id from a subnet's canister range leads to the request being routed to a node on the corresponding subnet and produce an error response there, while using an effective canister id that is in no subnet's canister ranges would cause an error response from the node receiving the request.
```
Comment (mraszyk) [inline spec/index.adoc:609]: "from the node receiving the request." -> which subnet is the request routed to in this case?

Comment (mraszyk) [reply spec/index.adoc]: I believe that the boundary node responses with 404 because it cannot route the request.

Comment (mraszyk) [reply spec/index.adoc]: Confirmed by @r-birkner

**Review by Dfinity-Bjoern — APPROVED** [review]
Thanks for the clarifications!

**Pull request merged**